gsdk/src/lib.rs

```diff
@@ -51,84 +68,38 @@ pub mod gp {
 pub type BlockNumber = u32;
 
-/// Gear api wrapper.
-#[derive(Clone)]
-pub struct Api {
-    /// How many times we'll retry when rpc requests failed.
-    pub retry: u16,
-    client: OnlineClient<GearConfig>,
+/// Information of gas
+#[derive(Clone, Debug, Decode, Encode, PartialEq, Eq, Serialize, Deserialize)]
+pub struct GasInfo {
+    /// Represents minimum gas limit required for execution.
+    pub min_limit: u64,
+    /// Gas amount that we reserve for some other on-chain interactions.
+    pub reserved: u64,
+    /// Contains number of gas burned during message processing.
+    pub burned: u64,
 }
 
-impl Api {
-    /// Create new API client.
-    pub async fn new(url: Option<&str>) -> Result<Self> {
-        Self::new_with_timeout(url, None).await
-    }
-
-    /// Create new API client with timeout.
-    pub async fn new_with_timeout(url: Option<&str>, timeout: Option<u64>) -> Result<Self> {
-        Ok(Self {
-            // Retry our failed RPC requests for 5 times by default.
-            retry: 5,
-            client: OnlineClient::from_rpc_client(Arc::new(RpcClient::new(url, timeout).await?))
-                .await?,
-        })
-    }
+/// Gear gas node id.
+pub type GearGasNodeId = GasNodeId<MessageId, ReservationId>;
 
-    /// Setup retry times and return the API instance.
-    pub fn with_retry(mut self, retry: u16) -> Self {
-        self.retry = retry;
-        self
-    }
+/// Gear gas node.
+pub type GearGasNode = GasNode<AccountId32, GearGasNodeId, u64>;
 
-    /// Subscribe all blocks
-    ///
-    /// # Parse events from blocks
-    ///
-    /// ```ignore
-    /// use gsdk::metadata::{Event, system};
-    ///
-    /// let api = Api::new(None).await?;
-    /// let mut blocks = api.blocks().await?;
-    ///
-    /// while let Some(Ok(events)) = blocks.next_events().await {
-    ///   for event in events {
-    ///     let ev = event.as_root_event::<Event>();
-    ///
-    ///     // Match events here.
-    ///     matches!(ev, Event::System(system::Event::ExtrinsicSuccess(_)));
-    ///   }
-    /// }
-    /// ```
-    pub async fn blocks(&self) -> Result<types::Blocks> {
-        Ok(types::Blocks(self.client.blocks().subscribe_all().await?))
-    }
+/// Gear pages.
+pub type GearPages = HashMap<u32, Vec<u8>>;
 
-    /// Subscribe finalized blocks
-    pub async fn finalized_blocks(&self) -> Result<types::Blocks> {
-        Ok(types::Blocks(
-            self.client.blocks().subscribe_finalized().await?,
-        ))
-    }
+/// Transaction in block.
+pub type InBlock = Result<TxInBlock<GearConfig, OnlineClient<GearConfig>>>;
 
-    /// New signer from api
-    pub fn signer(self, suri: &str, passwd: Option<&str>) -> Result<Signer> {
-        Signer::new(self, suri, passwd)
-    }
-}
-
-impl Deref for Api {
-    type Target = OnlineClient<GearConfig>;
-
-    fn deref(&self) -> &Self::Target {
-        &self.client
-    }
-}
+/// Transaction status.
+pub type TxStatus = tx::TxStatus<GearConfig, OnlineClient<GearConfig>>;
```

Comment:
```suggestion
pub type TxStatus = TxStatus<GearConfig, OnlineClient<GearConfig>>;
```

Reply:
the name of `TxStatus` is duplicated

Reply:
`GearTxStatus maybe`

Reply:
to specify underlying generic type

Reply:
makes sense, but since our library is just for our API, I think we can take the name of `TxStatus`, instead, `subxt::TxStatus` -> `SubxtTxStatus`